editoast/database/src/db_connection_pool.rs

```diff
@@ -364,64 +353,63 @@ impl DbConnectionPoolV2 {
         std::iter::repeat_with(|| self.get())
     }
 
-    #[cfg(not(feature = "testing"))]
-    pub async fn from_pool(pool: Arc<Pool<AsyncPgConnection>>) -> Self {
-        Self { pool }
-    }
-
-    #[cfg(feature = "testing")]
-    pub async fn from_pool(pool: Arc<Pool<AsyncPgConnection>>) -> Self {
-        Self::from_pool_test(pool, true).await
-    }
-
-    #[cfg(feature = "testing")]
-    pub async fn from_pool_test(pool: Arc<Pool<AsyncPgConnection>>, transaction: bool) -> Self {
-        use diesel_async::AsyncConnection as _;
-
-        let mut connection = pool
-            .get()
-            .await
-            .expect("cannot acquire a connection in the test pool");
-        connection.set_instrumentation(tracing_instrumentation::TracingInstrumentation::default());
-        if transaction {
-            connection
-                .begin_test_transaction()
-                .await
-                .expect("cannot begin a test transaction");
-        }
-        let test_connection = Some(DbConnection::new(Arc::new(RwLock::new(connection))));
-
-        Self {
+    #[cfg(any(test, feature = "testing"))]
+    async fn new_test(test_name: String) -> Result<Self, DatabaseOperationError> {
+        let osrd_pool: Arc<Pool<AsyncPgConnection>> = create_connection_pool(
```

Comment:
```suggestion
        let osrd_pool = Arc::new(create_connection_pool(
```

nit: more readable and doesn't require explicit typing